src/main/java/org/jenkinsci/plugins/docker/workflow/FromFingerprintStep.java

```diff
@@ -87,8 +89,9 @@ public String getToolName() {
             String fromImage = null;
             FilePath dockerfile = workspace.child(step.dockerfile);
             InputStream is = dockerfile.read();
+            BufferedReader r = null;
             try {
-                BufferedReader r = new BufferedReader(new InputStreamReader(is)); // encoding probably irrelevant since image/tag names must be ASCII
+                r = new BufferedReader(new InputStreamReader(is, "ASCII")); // encoding probably irrelevant since image/tag names must be ASCII
```

Comment:
So do you suggest using the same defaultEncoding as in the rest of the change? AFAIK that should cover whatever the system process spits out.